aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -99,29 +102,40 @@ static class DefaultAwsClientFactory implements AwsClientFactory {
     public S3Client s3() {
       return S3Client.builder()
           .httpClientBuilder(configureHttpClientBuilder(httpClientType))
-          .applyMutation(builder -> configureEndpoint(builder, s3Endpoint))
+          .applyMutation(builder -> configureRegion(builder, s3Endpoint))
+          .applyMutation(builder -> configureEndpoint(builder, region))
```

Comment:
This looks suspicious. It calls `configureRegion` with `s3Endpoint` and `configureEndpoint` with `region`.

Reply:
Sorry, my bad.